File: app-films-catalog-libs/common-rest/src/main/java/ru/micron/libs/dto/BasicResponse.java
package ru.micron.libs.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Builder.Default;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@Schema(description = "Basic response model")
public class BasicResponse<T> {

  @Schema(description = "Payload data", nullable = true)
  private T data;

  @Schema(description = "Indicates whether the processing result is successful or not")
  @Default
  private boolean success = true;

  @Schema(description = "Appears only if success is false", nullable = true)
  private ErrorObject error;

  @Schema(description = "indicates whether this response is deprecated or not")
  private boolean deprecated;
}

File: app-core/src/main/kotlin/ru/micron/dto/BasicResponse.kt
package ru.micron.dto

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Basic response model")
data class BasicResponse<T>(
    @Schema(description = "Payload data", nullable = true)
    var data: T? = null,
    @Schema(description = "Indicates whether the processing result is successful or not")
    var success: Boolean? = true,
    @Schema(description = "Appears only if success is false", nullable = true)
    var error: ErrorObject? = null,
    @Schema(description = "indicates whether this response is deprecated or not")
    var deprecated: Boolean? = false
) {
    constructor(data: T) : this(data, true)
    constructor(success: Boolean?) : this(null, success)
    constructor(data: T, success: Boolean?) : this(data, success, null)
}
